src/kzg/mod.rs

```diff
@@ -284,9 +286,13 @@ fn eval_poly(blob: &[Fr], z: &Fr) -> Fr {
                 el.mul_assign(&root);
                 let mut z_1 = z.clone();
                 z_1.sub_assign(&root);
-                el.mul_assign(&z_1.inverse().unwrap());
-                acc.add_assign(&el);
-                acc
+                if z_1 != Fr::zero() {
+                    el.mul_assign(&z_1.inverse().unwrap());
+                    acc.add_assign(&el);
+                    acc
+                } else {
+                    acc
```

Comment:
I think it's invalid. If you really want to handle unprobable edge case of hitting a domain, then you just check if Z is in domain and pick a particular value from `blob`, otherwise do as before